### ISSUE TITLE
Error in RecLayer construction when input is [B|F|T]

### ISSUE THREAD

**Issue by JackTemaki** [rwth-i6/returnn#380]
The network is not constructed correctly when using `"auto_use_channel_first": True` in the network.

The test case is here: https://github.com/rwth-i6/returnn/compare/master...JackTemaki:axes_optim_test
with the network:
```python3
  net_dict = {
    'conv0': {'L2': 1e-07, 'activation': 'relu', 'class': 'conv', 'filter_size': (5,), 'from': ['data'],
                    'n_out': 9, 'padding': 'same', 'auto_use_channel_first': True},
    'lstm0': {'class': 'rec',
                 'direction': 1,
                 'from': ['conv0'],
                 'n_out': 2,
                 'unit': 'nativelstm2',
                 },
    'output': {'class': 'copy', 'from': 'lstm0', 'target': 'data', 'loss': 'mse'}
  }
```

The error log can be found here: https://gist.github.com/JackTemaki/d0f407233bfd6149d41bc79431bca170,
in short:
```python3
File "/home/nick/github/returnn/returnn/tf/util/data.py", line 606, in sanity_check
    line: assert axis != self.batch_dim_axis, "%s: axis %s (%i) must be different from batch_dim_axis (%i)" % (
            self, axis_name, axis, self.batch_dim_axis)
    locals:
      axis = <local> 1
      self = <local> Data(name='lstm0_output', shape=(2, None), batch_dim_axis=1, feature_dim_axis=1, batch_shape_meta=[T|2,B|F,'time:var:extern_data:data'])
      self.batch_dim_axis = <local> 1
      axis_name = <local> 'feature_dim_axis', len = 16
AssertionError: Data(name='lstm0_output', shape=(2, None), batch_dim_axis=1, feature_dim_axis=1, batch_shape_meta=[T|2,B|F,'time:var:extern_data:data']): axis feature_dim_axis (1) must be different from batch_dim_axis (1)
```

I am assuming that there is something wrong in the template construction when the data input is in the shape `[B|F|T]` as returned from the `ConvLayer` in this test case. I will investigate later.

**Comment by albertz** [comment]
If the problem is the input shape [B,F,T], we can construct a simpler test case, where you just feed in such a tensor to a reclayer.

**Comment by JackTemaki** [comment]
The test case was originally intended to search for a different bug. I will add a different test case when submitting the PR with the fix.

**Comment by JackTemaki** [comment]
Fixed with #381
